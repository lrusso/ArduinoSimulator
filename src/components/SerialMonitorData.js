import React from "react"
import { useSimulatorContext } from "../contexts/SimulatorContext"

const SerialMonitorData = () => {
  const { outputData } = useSimulatorContext()
  const refData = React.useRef(null)

  React.useEffect(() => {
    if (refData.current) {
      refData.current.innerHTML = outputData
      refData.current.scrollIntoView(false)
    }
  }, [outputData])

  return (
    <div style={styles.container}>
      <div style={styles.data} ref={refData}></div>
    </div>
  )
}

const styles = {
  container: {
    position: "fixed",
    left: 0,
    right: 0,
    bottom: 0,
    height: "137px",
    overflow: "scroll",
    backgroundColor: "white",
  },
  data: {
    padding: "10px",
    whiteSpace: "pre-line",
    fontFamily: "monospace,monospace",
    overflowWrap: "break-word",
    WebkitUserSelect: "text",
    MozUserSelect: "text",
    userSelect: "text",
  },
}

export default SerialMonitorData
