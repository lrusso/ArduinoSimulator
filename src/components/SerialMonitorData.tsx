import React from "react"
import { useSimulatorContext } from "../contexts/SimulatorContext"

const SerialMonitorData = () => {
  const { outputData, simulatorRunning } = useSimulatorContext()
  const refData = React.useRef<HTMLDivElement>(null)

  React.useEffect(() => {
    if (refData.current) {
      refData.current.innerHTML = outputData
      refData.current.scrollIntoView(false)
    }
  }, [outputData])

  return (
    <div className="serialData">
      <div className={`serialData-inner ${!simulatorRunning && "disabled"}`}>
        <div className="serialData-data" ref={refData}></div>
      </div>
    </div>
  )
}

export default SerialMonitorData
