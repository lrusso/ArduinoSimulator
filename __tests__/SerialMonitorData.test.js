import React from "react"
import renderer from "react-test-renderer"
import SerialMonitorData from "../src/components/SerialMonitorData"

describe("SerialMonitorData", () => {
  test("Rendering 1", () => {
    const tree = renderer.create(<SerialMonitorData />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
