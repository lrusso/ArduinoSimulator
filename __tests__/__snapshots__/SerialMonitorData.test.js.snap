// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SerialMonitorData Rendering 1 1`] = `
<div
  style={
    {
      "backgroundColor": "white",
      "bottom": 0,
      "height": "137px",
      "left": 0,
      "overflow": "scroll",
      "position": "fixed",
      "right": 0,
    }
  }
>
  <div
    style={
      {
        "MozUserSelect": "text",
        "WebkitUserSelect": "text",
        "fontFamily": "monospace,monospace",
        "overflowWrap": "break-word",
        "padding": "10px",
        "userSelect": "text",
        "whiteSpace": "pre-line",
      }
    }
  />
</div>
`;
